### PR TITLE
Mac and compatibility fixes

### PR DESCRIPTION
**Pull request by neurolabusc**
Minor modifications for https://github.com/cloudflare/zlib/pull/6

### CODE REVIEW

File: deflate.c [hunk 1]
```diff
@@ -50,6 +50,9 @@
 /* @(#) $Id$ */
 
 #include "deflate.h"
+#ifndef UINT32_MAX
```
Comment (vkrasnov) [inline deflate.c]: Isn't the change for zconf.* sufficient?

File: contrib/amd64/crc32-pclmul_asm.S [hunk 2]
```diff
@@ -53,6 +55,17 @@ name:
 #define ENDPROC(name) \
 .size name, .-name
 
+#else // __APPLE__
```
Comment (vkrasnov) [inline contrib/amd64/crc32-pclmul_asm.S]: I merged it from the original PR

File: test/example.c [hunk 3]
```diff
@@ -7,6 +7,19 @@
...
+#ifndef UINT32_MAX
+#include <stdint.h>
+#endif
+
+/*support MSVC 2013 prior to https://github.com/libgit2/libgit2/issues/3476 */
+#if defined(_MSC_VER)
+ GIT_BEGIN_DECL
+# include "inttypes.h"
+ GIT_END_DECL
+/** This check is needed for importing this file in an iOS/OS X framework throws an error in Xcode otherwise.*/
```
Comment (vkrasnov) [inline test/example.c]: Comment format (this and the rest) is /* text */ specifically space before and after, don't start with a capital

File: test/example.c [hunk 4]
```diff
@@ -558,9 +571,11 @@ int main(argc, argv)
     } else if (strcmp(zlibVersion(), ZLIB_VERSION) != 0) {
         fprintf(stderr, "warning: different zlib version\n");
     }
-
-    printf("zlib version %s = 0x%04x, compile flags = 0x%llx\n",
-            ZLIB_VERSION, ZLIB_VERNUM, zlibCompileFlags());
+    /* ensure flags is 64 bit regardless whether uLong is 32 or 64 bit
+     */
+	uint64_t flags64 = zlibCompileFlags();
+    printf("zlib version %s = 0x%" PRIx64 ", compile flags = 0x%llx\n",
```
Comment (vkrasnov) [inline test/example.c]: Inconsistent. Need yo replace all %ll with PRI. I also assumed that %04x was OK?

File: test/example.c [hunk 5]
```diff
@@ -558,9 +571,11 @@ int main(argc, argv)
     } else if (strcmp(zlibVersion(), ZLIB_VERSION) != 0) {
         fprintf(stderr, "warning: different zlib version\n");
     }
-
-    printf("zlib version %s = 0x%04x, compile flags = 0x%llx\n",
-            ZLIB_VERSION, ZLIB_VERNUM, zlibCompileFlags());
+    /* ensure flags is 64 bit regardless whether uLong is 32 or 64 bit
+     */
+	uint64_t flags64 = zlibCompileFlags();
```
Comment (vkrasnov) [inline test/example.c]: Why?

**Comment by neurolabusc** [comment]
Not required with recent patches.

**Pull request closed without merging**